src/state/opamSwitchState.ml

```diff
@@ -278,8 +340,75 @@ let load lock_kind gt rt switch =
       conf_files
       OpamPackage.Set.empty
   in
+  (* depext check *)
+  let sys_packages =
+    if OpamStateConfig.(!r.ignore_depext || !r.depext_ignore_opam) then
+      OpamPackage.Map.empty
+    else
+      system_packages switch_config packages
+        ~depexts:(fun package ->
+            let env =
+              OpamPackageVar.resolve_switch_raw ~package gt switch switch_config
+            in
+            depexts_raw ~env package opams)
+  in
+  let sys_packages_changed =
+    if OpamStateConfig.(!r.ignore_depext || !r.depext_ignore_opam) then
+      OpamPackage.Set.empty
+    else
+    let sys_packages =
+      OpamPackage.Map.filter (fun pkg spkg ->
+          OpamPackage.Set.mem pkg installed
+          && not (OpamSysPkg.Set.is_empty spkg.OpamSysPkg.s_available
+                  && OpamSysPkg.Set.is_empty spkg.OpamSysPkg.s_not_found))
+        sys_packages
+    in
+    if OpamPackage.Map.is_empty sys_packages then
+      OpamPackage.Set.empty
+    else
+      (let lchanged = OpamPackage.Map.keys sys_packages in
+       let changed = OpamPackage.Set.of_list lchanged in
+       let sgl_pkg = OpamPackage.Set.cardinal changed = 1 in
+       let open OpamSysPkg.Set.Op in
+       let missing =
+         OpamPackage.Map.map (fun sys ->
+             sys.OpamSysPkg.s_available ++ sys.OpamSysPkg.s_not_found)
+           sys_packages
+       in
+       let sgl_spkg =
+         try
+           OpamSysPkg.Set.cardinal
+             (OpamPackage.Map.fold (fun _ sp acc ->
+                  let acc = acc ++ sp in
+                  if OpamSysPkg.Set.cardinal acc > 1 then
+                    raise Not_found
+                  else acc)
+                 missing OpamSysPkg.Set.empty) <=  1
+         with Not_found -> false
+       in
+       OpamConsole.error
+         "System package%s from which depends installed opam package%s no \
+          more installed on you system:\n%s\n \
+          %s %s been marked as removed, and opam will try to \
+          reinstall %s. You should reinstall system package first."
+         (if sgl_spkg then "" else "s")
+         (if sgl_pkg then " is" else "s are")
+         (OpamStd.Format.itemize (fun (pkg, spkg) ->
+              Printf.sprintf "%s: %s"
+                (OpamPackage.to_string pkg)
+                (OpamStd.Format.pretty_list spkg))
+             (OpamPackage.Map.bindings
+                (OpamPackage.Map.map (fun s ->
+                     List.map OpamSysPkg.to_string (OpamSysPkg.Set.elements s))
+                    missing)))
+         (OpamStd.Format.pretty_list (List.map OpamPackage.to_string lchanged))
+         (if sgl_pkg then "has" else "have")
+         (if sgl_pkg then "it" else "them")
+       ;
+       changed)
+  in
   let installed =
-    installed -- ext_files_changed
+    installed -- ext_files_changed -- sys_packages_changed
```

Comment:
This is not completely correct, but the issue already existed with `ext_files_changed`. I have pending work to fix both, so it's OK for now ;)

Reply:
yes, kept the old behaviour. When moving to the new one, these changes will also be updated